@types/interfaces.d.ts

```diff
@@ -0,0 +1,19 @@
+import { ContextMenuElement } from '../src/vaadin-context-menu.js';
+
+export interface ContextMenuItem {
+  text?: string,
+  component?: string | HTMLElement,
+  disabled?: boolean,
+  checkable?: boolean,
+  checked?: boolean,
+  children?: ContextMenuItem[]
+}
+
+export type ContextMenuRenderer = (
+  root: HTMLElement,
+  owner?: ContextMenuElement,
+  context?: {
+    target: HTMLElement;
+    detail?: object;
+  }
```

Comment:
Doesn't context-menu ensure that `owner` and `context` for the overlay are defined when the overlay calls `renderer`?

Reply:
I also think `owner` and `context` are always set or @web-padawan can you think of any situation where they won't be set when this renderer is called?

Also the JSDoc for `renderer` property  calls this `owner` with name `contextMenu`. Either change it to `contextMenu` here too or update the JSDoc to also talk about `owner`.

See also https://github.com/vaadin/vaadin-overlay/blob/master/%40types/interfaces.d.ts

Reply:
I know that these are present internally, but user does not have to provide them. Neither `contextMenu` nor `context` are not mandatory, so the following is possible:

```js
contextMenu.renderer = function(root) {
  // render stuff
}
```

Reply:
Changed `owner` to `contextMenu`.

Reply:
Afaik the method type basically declares what parameters the function will (or should) be called with. It doesn't care if the function uses all those parameters.

See also https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#optional-parameters-in-callbacks

Reply:
So overriding the callback with a function that only takes one parameter is perfectly valid even if the other params aren't declared as optional.

Reply:
So the current version looks like this
```ts
export type ContextMenuRenderer = (
  root: HTMLElement,
  contextMenu?: ContextMenuElement,
  context?: ContextMenuRendererContext
) => void;
```
I think it should be fine to remove those question marks.

Everything else in the PR is currently ok to me.

Reply:
Thanks, done.